state-sync/src/coordinator.rs

```diff
@@ -914,71 +977,89 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             new_version
         );
 
-        // The overall chunk processing duration is calculated starting from the very first attempt
-        // until the commit
-        if let Some(first_attempt_tst) = self.request_manager.get_first_request_time(known_version)
-        {
-            if let Ok(duration) = SystemTime::now().duration_since(first_attempt_tst) {
-                counters::SYNC_PROGRESS_DURATION.observe_duration(duration);
+        // Log the request processing time (time from first requested until now).
+        match self.request_manager.get_first_request_time(known_version) {
+            None => {
+                info!(
+                    LogSchema::event_log(LogEntry::ProcessChunkResponse, LogEvent::ReceivedChunkWithoutRequest),
+                    "Received a chunk of size {}, without making a request! Previous version: {}, new version {}",
+                    chunk_size,
+                    known_version,
+                    new_version
+                );
+            }
+            Some(first_request_time) => {
+                if let Ok(duration) = SystemTime::now().duration_since(first_request_time) {
+                    counters::SYNC_PROGRESS_DURATION.observe_duration(duration);
+                }
```

Comment:
This is another one, where I think just putting an `else` at the end of the `if let` would have been fine.  The case is if `Some` and the other case is `None` so no arguments are needed from it.

Reply:
Yeah, I understand they are equivalent.  Here's just the problem with making changes like this.

Engineer A, has personal preference to have `if let`, Engineer B, has personal preference to have `match`.  Whenever Engineer A decides to touch the code, they determine they want to change it to the way they like it, and whenever Engineer B decides to touch the code, they determine they want to change it to the way like it.  Leads to some churn back and forth (which I've seen happen before not necessarily here).

I don't particularly have an issue with any of these, I'm just pointing them out to make sure you focus on the content, and not the format (unless it's egregious).  Plus it erases the original history of changes with the refactor history, which makes it harder to determine why a certain change was made.

Reply:
That's fair. Given that I'm probably the only one who will be working on this for a while, I've been pretty relaxed at changing things to how I like to see them, but I appreciate that there are differences in taste when it comes to these things 😄 Will try to be conscious of them going forward.

Reply:
I think for options if let is nicer... also match causes an extra layer of nesting, nesting is evil...